http/file_server_test.ts

```diff
@@ -1622,3 +1622,36 @@ Deno.test(
     }
   },
 );
+
+Deno.test("serveDir serves 404.html when resource is not found", async () => {
+  const url = "http://localhost:4507/not-found.html";
+  const expectedText = "This-is-the-content-of-the-fallback-file/404.html";
+  const res = await serveDir(new Request(url), {
+    fsRoot: "./http/testdata",
+  });
+  assertEquals(res.status, 200);
```

Comment:
I think the http status should be still 404 when serving 404.html.

I think we don't need to use `serveFile` for returning the contents of 404.html. I think we can do something like `new Response((await Deno.open(path)).readable, { status: 404 })`